design/existing-resource-policy_design.md

```diff
@@ -0,0 +1,247 @@
+1. `none`: This is the existing behavior, if Velero encounters a resource that already exists in the cluster, we simply
+skip restoration.
+2. `update` or `merge` or `patch`: In this behaviour Velero will try an attempt to patch the resource with the backed up version, 
+if the patch fails log it as a warning and continue the restore process, user may wish specify the merge type.
+3. `updateAll`: Similar to `update` option but Velero will update labels on unchanged resources too.
+4. `recreate`:  If resource already exists, then Velero will delete it and recreate the resource.
+
+In summary,
+
+| Policy      | Action performed by Velero                                                              | Velero Logging                                                |
+|-------------|-----------------------------------------------------------------------------------------|---------------------------------------------------------------|
+| `none`      | Existing Velero behavior.                                                               | `info` log for unchanged resource, `warn` on changed resource |
+| `update`    | Velero tries to attempt a `patch` on changed resource                                   | `info` log for unchanged, `warn` if patch failed              |
+| `updateAll` | Velero updates labels on unchanged resource  and attempts a `patch` on changed resource | `info` log for unchanged and `warn` if patch failed           |
+
+*Note:* The `recreate` option is a non-goal for this enhancement proposal, but it is considered as a future scope.
+Another thing to highlight is that Velero will not be deleting any resources in any of the policy options proposed in 
+this design but Velero will annotate the changed resources in `updateAll` policy option.
+
+Example:
+A. The following Restore will execute the `existingResourcePolicy` restore type `none` for the `services` and `deployments` present in the `velero-protection` namespace.
+
+```
+Kind: Restore
+
+…
+
+includeNamespaces: velero-protection
+includeResources:
+  - services
+  - deployments
+existingResourcePolicy: none
+
+```
+
+B. The following Restore will execute the `existingResourcePolicy` restore type `update` for the `secrets` and `daemonsets` present in the `gdpr-application` namespace.
+```
+Kind: Restore
+
+…
+includeNamespaces: gdpr-application
+includeResources:
+  - secrets
+  - daemonsets
+existingResourcePolicy: update
+```
+
+### Approach 2: Add a new spec field `existingResourcePolicyConfig` to the Restore API
+In this approach we give user the ability to specify which resources are to be included for a particular kind of force update behaviour, essentially a more granular approach where in the user is able to specify a resource:behaviour mapping. It would look like:
+`existingResourcePolicyConfig`:
+- `patch:` 
+    - `includedResources:` [ ]string
+- `recreate:` [ ]string
+    - `includedResources:` [ ]string
```

Comment:
What if the values of `includeResources` of different actions have overlap?

Reply:
We would need to be explicit in documenting this, but I think "first match" is good enough. Ideally there would be no overlap here, but if there is, take the first one in the list.

Reply:
Or error out so user fix it in the next run?